contracts/AJMarket.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+/// @custom:security-contact security@ajcoin.xyz
+contract AJMarket is Ownable {
+    using Counters for Counters.Counter;
+
+    IERC20 ajCoin;
```

Comment:
I'm a bit concerned about the visibility into the code in block explorers, which I feel like a proxy sometimes unintentionally obfuscates.  It's a bit of an assumption tho.  If you have more info on this I'm interested because I've gone back and forth.

The nice thing about this use case is I can create a new contract and put new listings on the new contract. while I depreciate the old one.

Reply:
Fair point. From what I have seen, every proxy extension has a publicly readable contract for the implementation address, which then can be listened for events - have noticed that etherscan and even polygonscan have capability to determine if its a proxy contract and point to the actual implementation. Here's an example of a proxy contract on etherscan - there are tabs for Read/Write as Proxy.

Agreed, its just a bit of maintenance "overhead" on your end to deploy a new contract and update it with the listings.

Reply:
I'll skip it for this project, but thanks for the info, I have been wondering about this.  I don't see the link tho.  Can you ping me it on slack when you have a chance?